crates/sui-types/src/effects/effects_v2.rs

```diff
@@ -0,0 +1,271 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::base_types::{EpochId, ObjectDigest, ObjectRef, TransactionDigest};
+use crate::digests::TransactionEventsDigest;
+use crate::effects::{TransactionEffectsAPI, TransactionEffectsDebugSummary};
+use crate::execution_status::ExecutionStatus;
+use crate::gas::GasCostSummary;
+use crate::object::Owner;
+use crate::storage::{DeleteKind, WriteKind};
+use crate::{ObjectID, SequenceNumber};
+use serde::{Deserialize, Serialize};
+use std::collections::HashSet;
+
+/// The response from processing a transaction or a certified transaction
+#[derive(Eq, PartialEq, Clone, Debug, Serialize, Deserialize)]
+pub struct TransactionEffectsV2 {
+    /// The status of the execution
+    status: ExecutionStatus,
+    /// The epoch when this transaction was executed.
+    executed_epoch: EpochId,
+    gas_used: GasCostSummary,
+    /// The transaction digest
+    transaction_digest: TransactionDigest,
+    /// The updated gas object reference. Have a dedicated field for convenient access.
+    /// It's also included in `changed_objects`.
+    gas_object: (ObjectRef, Owner),
```

Comment:
Why not make this like a u16 index into `changed_objects`?

Reply:
Good idea!